### PR TITLE
macOS file operations improvements

### PR DESCRIPTION
**Pull request by marcograss**
fixes for #292 #236

### CODE REVIEW

File: src/CutterApplication.h [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+
+class CutterApplication : public QApplication
+{
+    Q_OBJECT
+    Q_PROPERTY(MainWindow* mainWindow READ mainWindow WRITE setMainWindow)
+public:
+    CutterApplication(int &argc, char **argv) : QApplication(argc, argv){
+        setOrganizationName("cutter");
+        setApplicationName("cutter");
```
Comment (xarkes) [inline src/CutterApplication.h]: Uppercase for both would be better

File: src/CutterApplication.h [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        // Check r2 version
+        QString r2version = r_core_version();
+        QString localVersion = "" R2_GITTAP;
+        if (r2version != localVersion)
+        {
+            QMessageBox msg;
+            msg.setIcon(QMessageBox::Critical);
+            msg.setStandardButtons(QMessageBox::Yes | QMessageBox::No);
+            msg.setWindowTitle(QObject::tr("Version mismatch!"));
+            msg.setText(QString(QObject::tr("The version used to compile cutter (%1) does not match the binary version of radare2 (%2). This could result in unexpected behaviour. Are you sure you want to continue?")).arg(localVersion, r2version));
```
Comment (xarkes) [inline src/CutterApplication.h]: Uppercase Cutter

**Comment by xarkes** [comment]
I am not very found of putting code in a header file but it looks cleaner anyway. Thoughts @thestr4ng3r?
Can you just fix the application name? Thanks, looks good to me.

**Comment by thestr4ng3r** [comment]
Yes, only short methods with a bigger benefit of inlining like getters and setters should be in the header.
The rest looks good to me too.

**Comment by xarkes** [comment]
Thank you!

**Comment by Maijin** [comment]
Dock  drag and drop doesn't work

**Comment by marcograss** [comment]
@Maijin can you give me a hint how to repro? I have it on my dock and i tried with multiple files, I'm on 10.13.3

**Comment by Maijin** [comment]
10.13.3 (17D47) - Tried dock with the app located in Applications and in the build

**Pull request merged**